# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/blindcoder/ksimus/gcc34.patch.ksimus
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Courtesy of debian.
Make ksimus compile with gcc34

diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/boolean/boolean1out.cpp ./ksimus/boolean/boolean1out.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/boolean/boolean1out.cpp	2002-12-22 00:30:04.000000000 +0100
+++ ./ksimus/boolean/boolean1out.cpp	2004-08-13 09:09:22.352099113 +0200
@@ -97,12 +97,12 @@
 bool Boolean1Out::getResetState() const
 {
 	return m_flags & FLAGS_RESET_TRUE;
-};
+}
 	
 bool Boolean1Out::getResetStateInit() const
 {
 	return m_flags & FLAGS_RESET_INIT_TRUE;
-};
+}
 	
 
 
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/boolean/booleanled.cpp ./ksimus/boolean/booleanled.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/boolean/booleanled.cpp	2002-10-27 00:55:28.000000000 +0200
+++ ./ksimus/boolean/booleanled.cpp	2004-08-13 09:09:44.947664065 +0200
@@ -422,7 +422,7 @@
 	{
 		return m_color;
 	}
-};
+}
 
 
 void BooleanLedColorPropertyWidget::slotRed()
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/component.cpp ./ksimus/component.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/component.cpp	2003-02-16 17:32:38.000000000 +0100
+++ ./ksimus/component.cpp	2004-08-13 09:19:33.145244441 +0200
@@ -522,7 +522,7 @@
 		return getDefaultName();
 	}
 	return m_name;
-};
+}
 
 void Component::setName(const QString & newName)
 {
@@ -537,14 +537,14 @@
 		m_name = newName.simplifyWhiteSpace();
 	}
 	emit signalSetName(getName());
-};
+}
 
 QString Component::getDefaultName() const
 {
 	return QString::fromLatin1("%1 %2")
 	                           .arg(getInfo()->getName())
 	                           .arg(getSerialNumber());
-};
+}
 
 bool Component::hasDefaultName() const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/compview.cpp ./ksimus/compview.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/compview.cpp	2003-02-16 17:33:46.000000000 +0100
+++ ./ksimus/compview.cpp	2004-08-13 09:19:15.721893193 +0200
@@ -278,14 +278,14 @@
 eViewType CompView::getViewType() const
 {
 	return m_p->viewType;
-};
+}
 
 /** Change the type of view
    	@see enum eViewType */
 void CompView::setViewType(eViewType newType)
 {
 	m_p->viewType = newType;
-};
+}
 
 
 /** Manage the mouse pressed event if required */
@@ -689,7 +689,7 @@
 			module->getUserView()->setViewChanged(true);
 		}
 	}
-};
+}
 
 
 /** Creates a new Widget  */
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/componentlayout.cpp ./ksimus/items/componentlayout.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/componentlayout.cpp	2002-12-29 15:55:06.000000000 +0100
+++ ./ksimus/items/componentlayout.cpp	2004-08-13 09:16:38.996719041 +0200
@@ -303,7 +303,7 @@
 	{
 		return m_parent.layout;
 	}
-};
+}
 
 ComponentLayout * CompLayoutBase::getBaseLayout() const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/connectorbase.cpp ./ksimus/items/connectorbase.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/connectorbase.cpp	2003-05-20 00:02:22.000000000 +0200
+++ ./ksimus/items/connectorbase.cpp	2004-08-13 09:17:58.931567097 +0200
@@ -889,7 +889,7 @@
 	const QPoint & pos, ConnOrientationType orient, const ConnectorInfo * ci)
 	:	ConnectorBase (comp,name,i18nName,pos,orient,CD_INPUT,ci)
 {
-};
+}
 
 /** Checks the connector
 *   eg. if input is connected.
@@ -918,7 +918,7 @@
 			const QPoint & pos, ConnOrientationType orient, const ConnectorInfo * ci)
 	:	ConnectorBase (comp,name,i18nName,pos,orient,CD_OUTPUT,ci)
 {
-};
+}
 
 //**************************************************************************	
 // *** class ConnectorTristateBase ***
@@ -930,7 +930,7 @@
 	:	ConnectorBase (comp,name,i18nName,pos,orient,CD_TRISTATE,ci),
 		m_outActive(false)
 {
-};
+}
 
 void ConnectorTristateBase::reset()
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/connectorbooltristate.cpp ./ksimus/items/connectorbooltristate.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/connectorbooltristate.cpp	2003-02-28 00:06:19.000000000 +0100
+++ ./ksimus/items/connectorbooltristate.cpp	2004-08-13 09:12:31.345367769 +0200
@@ -139,7 +139,7 @@
 	{
 		return m_outData.getTristate();
 	}
-};
+}
 
 /** Return the current wire state */
 KSimBoolTristate ConnectorBoolTristate::getInput() const
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/externalconnector.cpp ./ksimus/items/externalconnector.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/externalconnector.cpp	2002-11-24 02:59:45.000000000 +0100
+++ ./ksimus/items/externalconnector.cpp	2004-08-13 09:16:52.830615969 +0200
@@ -406,13 +406,13 @@
 void ExternalConnector::setPixmapOrientation(ConnOrientationType orientation)
 {
 	m_pixmapOrient = orientation;
-};
+}
 		
 /** Sets the orientation of the connector in user view mode. */
 void ExternalConnector::setUserViewOrientation(ConnOrientationType orientation)
 {
 	m_userViewOrient = orientation;
-};
+}
 
 //##########################################################################################
 //##########################################################################################
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/watchitembase.cpp ./ksimus/items/watchitembase.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/watchitembase.cpp	2002-09-27 21:28:56.000000000 +0200
+++ ./ksimus/items/watchitembase.cpp	2004-08-13 09:16:25.370790497 +0200
@@ -162,7 +162,7 @@
 {
 	m_breakEnabled = ena;
 	showBreakCount(getBreaksActual());
-};
+}
 
 QString WatchItemBase::getWatchName() const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/watchitembooltristate.cpp ./ksimus/items/watchitembooltristate.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/watchitembooltristate.cpp	2003-02-28 00:06:19.000000000 +0100
+++ ./ksimus/items/watchitembooltristate.cpp	2004-08-13 09:12:11.278418409 +0200
@@ -171,7 +171,7 @@
 void WatchItemBoolTristateBase::ActivePropertyWidget::setMaxValue(unsigned int maxValue)
 {
 	m_spin->setMaxValue(maxValue);
-};
+}
 
 WatchItemBoolTristateBase::ActiveProperty WatchItemBoolTristateBase::ActivePropertyWidget::value() const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/wireproperty.cpp ./ksimus/items/wireproperty.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/wireproperty.cpp	2003-02-28 00:06:19.000000000 +0100
+++ ./ksimus/items/wireproperty.cpp	2004-08-13 09:17:39.531516353 +0200
@@ -339,7 +339,7 @@
 	:	WirePropertyInvalidBase(wire, getWirePropertyInvalidDifferentOutputsInfo())
 {
 	logWarning(getErrorMsg());
-};
+}
 
 /*WirePropertyInvalidDifferentOutputs::~WirePropertyInvalidDifferentOutputs()
 {
@@ -398,7 +398,7 @@
 	:	WirePropertyInvalidBase(wire, getWirePropertyInvalidDifferentOutputsInfo())
 {
 	logWarning(getErrorMsg());
-};
+}
 
 /*WirePropertyInvalidIncompatibleInputs::~WirePropertyInvalidIncompatibleInputs()
 {
@@ -457,7 +457,7 @@
 	:	WirePropertyInvalidBase(wire, getWirePropertyInvalidDifferentInputsNoOutputInfo())
 {
 	logWarning(getErrorMsg());
-};
+}
 
 /*WirePropertyInvalidDifferentInputsNoOutput::~WirePropertyInvalidDifferentInputsNoOutput()
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/items/wirepropertybooltristate.h ./ksimus/items/wirepropertybooltristate.h
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/items/wirepropertybooltristate.h	2003-02-28 00:06:19.000000000 +0100
+++ ./ksimus/items/wirepropertybooltristate.h	2004-08-13 09:11:50.406591409 +0200
@@ -73,27 +73,27 @@
 {
 	m_trueCount = trueCount;
 	m_falseCount = falseCount;
-};
+}
 
 inline void WireStateBoolTristate::setTrue(unsigned int trueCount)
 {
 	m_trueCount = trueCount;
-};
+}
 
 inline void WireStateBoolTristate::setFalse(unsigned int falseCount)
 {
 	m_falseCount = falseCount;
-};
+}
 
 inline unsigned int WireStateBoolTristate::getTrue() const
 {
 	return m_trueCount;
-};
+}
 
 inline unsigned int WireStateBoolTristate::getFalse() const
 {
 	return m_falseCount;
-};
+}
 
 inline bool WireStateBoolTristate::isTrue() const
 {
@@ -108,12 +108,12 @@
 inline bool WireStateBoolTristate::isActive() const
 {
 	return (getTrue() != 0) || (getFalse() != 0);
-};
+}
 
 inline bool WireStateBoolTristate::isInactive() const
 {
 	return (getTrue() == 0) && (getFalse() == 0);
-};
+}
 
 inline void WireStateBoolTristate::setTristate(KSimBoolTristate state)
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimdata.cpp ./ksimus/ksimdata.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimdata.cpp	2003-02-16 17:34:32.000000000 +0100
+++ ./ksimus/ksimdata.cpp	2004-08-13 09:18:18.327618449 +0200
@@ -70,7 +70,7 @@
 
 //#########################################################################################
 
-EnumDict<KSimData::eVersionType>::tData EnumDict<KSimData::eVersionType>::data[]
+template <> EnumDict<KSimData::eVersionType>::tData EnumDict<KSimData::eVersionType>::data[]
       = { {"error",        KSimData::versionError},
           {"as is",        KSimData::versionAsIs},
           {"unknown",      KSimData::versionUnknown},
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimeditor.cpp ./ksimus/ksimeditor.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimeditor.cpp	2003-05-20 00:03:24.000000000 +0200
+++ ./ksimus/ksimeditor.cpp	2004-08-13 09:18:55.533962225 +0200
@@ -119,7 +119,7 @@
 	CHECK_PTR(m_wireCursor);
 
 	setNow(arrowCursor);
-};
+}
 
 void KSimEditor::Cursor::set(KSimCursorType newCursor)
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimus.cpp ./ksimus/ksimus.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimus.cpp	2003-05-01 18:11:55.000000000 +0200
+++ ./ksimus/ksimus.cpp	2004-08-13 09:20:15.383823201 +0200
@@ -75,7 +75,7 @@
 //################################################################################
 
 
-EnumDict<eAppViewType>::tData EnumDict<eAppViewType>::data[]
+template <> EnumDict<eAppViewType>::tData EnumDict<eAppViewType>::data[]
       = { {"Sheet View",   APP_SHEET_VIEW},
           {"User View",    APP_USER_VIEW},
           {"Module View",  APP_MODULE_VIEW},
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimusdoc.cpp ./ksimus/ksimusdoc.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/ksimusdoc.cpp	2002-12-23 15:23:33.000000000 +0100
+++ ./ksimus/ksimusdoc.cpp	2004-08-13 09:19:59.394253985 +0200
@@ -127,7 +127,7 @@
 KSimusApp * KSimusDoc::getApp() const
 {
 	return (KSimusApp*)parent();
-};
+}
 
 void KSimusDoc::addView(KSimusView *view)
 {
@@ -541,13 +541,13 @@
 		m_modified=_m;
 	    getApp()->slotSaveAllowed(_m);
 	}
-};
+}
 
 /** returns if the document is modified or not. Use this to determine if your document needs saving by the user on closing.*/
 bool KSimusDoc::isModified() const
 {
 	return m_modified;
-};
+}
 
 void KSimusDoc::deleteContents()
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/lib/componentinfo.cpp ./ksimus/lib/componentinfo.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/lib/componentinfo.cpp	2002-08-05 23:43:32.000000000 +0200
+++ ./ksimus/lib/componentinfo.cpp	2004-08-13 09:11:07.939047457 +0200
@@ -32,7 +32,7 @@
 //##########################################################
 
 
-EnumDict<ViewAttribute>::tData EnumDict<ViewAttribute>::data[]
+template <> EnumDict<ViewAttribute>::tData EnumDict<ViewAttribute>::data[]
 			= { {"VA_SHEETVIEW", 			VA_SHEETVIEW},
           {"VA_USERVIEW",  			VA_USERVIEW},
 					{"VA_SHEET_AND_USER", VA_SHEET_AND_USER},
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/lib/packageinfo.cpp ./ksimus/lib/packageinfo.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/lib/packageinfo.cpp	2002-09-27 21:31:01.000000000 +0200
+++ ./ksimus/lib/packageinfo.cpp	2004-08-13 09:10:35.295010105 +0200
@@ -61,26 +61,26 @@
 const char * PackageInfo::getPackageVersion() const
 {
 	return m_packageVersion;
-};
+}
 
 const ComponentInfoList & PackageInfo::getComponentList() const
 {
 	return m_componentList;
-};
+}
 
 const ConnectorInfoList & PackageInfo::getConnectorList() const
 {
 	return m_connectorList;
-};
+}
 
 const WirePropertyInfoList & PackageInfo::getWirePropertyList() const
 {
 	return m_wirePropertyList;
-};
+}
 
 const ImplicitConverterInfoList & PackageInfo::getImplicitConverterList() const
 {
 	return m_implicitConverterList;
-};
+}
 
 
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/library.cpp ./ksimus/library.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/library.cpp	2003-02-28 00:06:18.000000000 +0100
+++ ./ksimus/library.cpp	2004-08-13 09:18:39.690370817 +0200
@@ -617,10 +617,10 @@
 const QStringList & Library::getInfoMessages() const
 {
 	return m_p->m_infoMessages;
-};
+}
 
 const QStringList & Library::getErrorMessages() const
 {
 	return m_p->m_errorMessages;
-};
+}
 
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/module/moduledata.cpp ./ksimus/module/moduledata.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/module/moduledata.cpp	2002-12-29 15:58:37.000000000 +0100
+++ ./ksimus/module/moduledata.cpp	2004-08-13 09:07:17.260115985 +0200
@@ -68,7 +68,7 @@
 static const char * sPixmapStore	= "Pixmap Store";
 static const char * sPixmapData = "Pixmap Data";
 
-EnumDict<ModuleViewType>::tData EnumDict<ModuleViewType>::data[]
+template <> EnumDict<ModuleViewType>::tData EnumDict<ModuleViewType>::data[]
       = { {"Generic",    MV_GENERIC},
           {"User View",  MV_USERVIEW},
           {"Pixmap",     MV_PIXMAP},
@@ -82,7 +82,7 @@
 }
 
 
-EnumDict<ModulePixmapStoreType>::tData EnumDict<ModulePixmapStoreType>::data[]
+template <> EnumDict<ModulePixmapStoreType>::tData EnumDict<ModulePixmapStoreType>::data[]
       = { {"Absolute",         MPS_ABSOLTUE},
           {"Relative Module",  MPS_RELATIVE_MODULE},
           {"Internal",         MPS_INTERNAL},
@@ -148,7 +148,7 @@
 const QSize & ModuleData::getGenericSize() const
 {
 	return m_genericSize;
-};
+}
 	
 PointList * ModuleData::getGenericConnPos()
 {
@@ -159,12 +159,12 @@
 		m_connPosGeneric->setAutoDelete(true);
 	}
 	return m_connPosGeneric;
-};
+}
 
 const QSize & ModuleData::getPixmapSize() const
 {
 	return m_pixmapSize;
-};
+}
 	
 PointList * ModuleData::getPixmapConnPos()
 {
@@ -175,7 +175,7 @@
 		m_connPosPixmap->setAutoDelete(true);
 	}
 	return m_connPosPixmap;
-};
+}
 
 OrientList * ModuleData::getPixmapConnOrient()
 {
@@ -186,7 +186,7 @@
 		m_connOrientPixmap->setAutoDelete(true);
 	}
 	return m_connOrientPixmap;
-};
+}
 
 /** returns a pointer to the pixmap
 	Creates an empty pixmap, if not exist */
@@ -280,7 +280,7 @@
 		m_connPosUserView->setAutoDelete(true);
 	}
 	return m_connPosUserView;
-};
+}
 
 OrientList * ModuleData::getUserViewConnOrient()
 {
@@ -291,7 +291,7 @@
 		m_connOrientUserView->setAutoDelete(true);
 	}
 	return m_connOrientUserView;
-};
+}
 
 
 /** Search externals in compList */
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/enumdict.h ./ksimus/tools/enumdict.h
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/enumdict.h	2001-11-24 02:11:32.000000000 +0100
+++ ./ksimus/tools/enumdict.h	2004-08-13 09:01:10.623853169 +0200
@@ -136,7 +136,7 @@
 const T * EnumDict<T>::find(const char * name) const
 {
 	return (const T *)EnumBaseDict::find(name);
-};
+}
 template<class T>
 T EnumDict<T>::find(const char * name, T defaultValue) const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimbaseuint.cpp ./ksimus/tools/ksimbaseuint.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimbaseuint.cpp	2002-08-05 23:44:00.000000000 +0200
+++ ./ksimus/tools/ksimbaseuint.cpp	2004-08-13 09:02:07.652183545 +0200
@@ -39,13 +39,13 @@
 		m_base(Decimal)
 {	
 	setText(text);
-};
+}
 
 
 void KSimBaseUInt::setText(const QString & str, bool * ok)
 {
 	m_value = convert(str, &m_base, ok);
-};
+}
 
 
 QString KSimBaseUInt::text() const
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimbooltristate.cpp ./ksimus/tools/ksimbooltristate.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimbooltristate.cpp	2003-02-28 00:06:19.000000000 +0100
+++ ./ksimus/tools/ksimbooltristate.cpp	2004-08-13 09:00:57.443856833 +0200
@@ -32,7 +32,7 @@
 
 //#########################################################################################
 
-EnumDict<eKSimBoolTristate>::tData EnumDict<eKSimBoolTristate>::data[]
+template <> EnumDict<eKSimBoolTristate>::tData EnumDict<eKSimBoolTristate>::data[]
       = { {"False",        KSIMBOOLTRISTATE_FALSE},
           {"True",         KSIMBOOLTRISTATE_TRUE},
           {"Inactive",     KSIMBOOLTRISTATE_INACTIVE},
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimdoubleedit.cpp ./ksimus/tools/ksimdoubleedit.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimdoubleedit.cpp	2002-02-04 23:47:31.000000000 +0100
+++ ./ksimus/tools/ksimdoubleedit.cpp	2004-08-13 09:04:37.611386281 +0200
@@ -47,12 +47,12 @@
 
 KSimDoubleEditValidator::KSimDoubleEditValidator(KSimDoubleEdit * parent, const char *name)
 	:	QDoubleValidator(-HUGE_VAL, HUGE_VAL, 10, parent, name)
-{};
+{}
 
 KSimDoubleEditValidator::KSimDoubleEditValidator(double bottom, double top, int decimals,
                                                  KSimDoubleEdit * parent, const char *name )
 	:	QDoubleValidator(bottom, top, decimals, parent, name)
-{};
+{}
 
 void KSimDoubleEditValidator::fixup(QString & input) const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimfilename.cpp ./ksimus/tools/ksimfilename.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimfilename.cpp	2002-09-27 21:33:15.000000000 +0200
+++ ./ksimus/tools/ksimfilename.cpp	2004-08-13 09:01:49.026015153 +0200
@@ -46,7 +46,7 @@
 
 static EnumDict<KSimFilename::ePathType> pathTypeDict;
 
-EnumDict<KSimFilename::ePathType>::tData EnumDict<KSimFilename::ePathType>::data[]
+template <> EnumDict<KSimFilename::ePathType>::tData EnumDict<KSimFilename::ePathType>::data[]
 			= { {"Absolute",          KSimFilename::PATH_ABSOLUTE},
 					{"Relative Document", KSimFilename::PATH_RELATIVE_DOCUMENT},
 					{"Relative Module",   KSimFilename::PATH_RELATIVE_MODULE},
@@ -539,27 +539,27 @@
 const KSimFilename & KSimFilenameWidget::getFileInfo() const
 {
 	return m_p->m_filename;
-};
+}
 
 KSimFilename & KSimFilenameWidget::getFileInfo()
 {
 	return m_p->m_filename;
-};
+}
 
 KSimLineEdit * KSimFilenameWidget::getFilenameEdit()
 {
 	return m_p->m_filenameEdit;
-};
+}
 
 QPushButton * KSimFilenameWidget::getFilenameDialogButton()
 {
 	return m_p->m_filenameDialogButton;
-};
+}
 
 QComboBox * KSimFilenameWidget::getPathTypeBox()
 {
 	return m_p->m_pathTypeBox;
-};
+}
 
 QString KSimFilenameWidget::getFileFilter() const
 {
diff -urN ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimtimebase.cpp ./ksimus/tools/ksimtimebase.cpp
--- ../tmp-orig/ksimus-0.3.6-2/ksimus/tools/ksimtimebase.cpp	2002-08-05 23:44:00.000000000 +0200
+++ ./ksimus/tools/ksimtimebase.cpp	2004-08-13 09:02:49.535816265 +0200
@@ -31,7 +31,7 @@
 #include "ksimtimeserver.h"
 #include "enumdict.h"
 
-EnumDict<eTimeUnit>::tData EnumDict<eTimeUnit>::data[]
+template <> EnumDict<eTimeUnit>::tData EnumDict<eTimeUnit>::data[]
       = { {"Ticks",    unit_ticks},
           {"1/10ns",   unit_tenth_nsec},
           {"1/10nsec", unit_tenth_nsec},



